test/parallel_api/experimental/for_loop.pass.cpp

```diff
@@ -314,9 +313,8 @@ test_for_loop_strided()
         ::std::vector<size_t> strides = {1, 2, 10, n > 1 ? n - 1 : 1, n > 0 ? n : 1, n + 1};
         for (size_t stride : strides)
         {
-            // for_loop staff is implemented for the host policies only
-            invoke_on_all_host_policies()(test_for_loop_strided_impl(), in_out.begin(), in_out.end(), expected.begin(),
-                                          expected.end(), in_out.size(), stride);
+            invoke_on_all_policies<>()(test_for_loop_strided_impl(), in_out.begin(), in_out.end(), expected.begin(),
```

Comment:
Due to onePDL has not hetero implementation for "for_loop", I would suggest:

1. testing just the hots policies  - `invoke_on_all_host_policies`
2. To add a static assert in `for_loop` implementation in case of hetero policy.

Reply:
This variant already has been implemented in the current PR earlier and rolled back.
But this mean that our users may have new compile errors if they have calls of `for_loop` with hetero-policies.